### PR TITLE
Follow up to 3855

### PR DESCRIPTION
**Pull request by mburke5678**
@gabemontero I might have gotten a bit carried away when editing. Can you review my changes to make sure I didn't change anything technical?

See preview:
http://file.rdu.redhat.com/~mburke/3855/build_strategies.html#pipeline-strategy-options

### CODE REVIEW

File: dev_guide/builds/build_strategies.adoc [hunk 1]
```diff
@@ -568,24 +567,25 @@ jenkinsPipelineStrategy:
...
+* You can add additional parameters (whose names differ from the names of the environment variables in the `BuildConfig`) 
+to the Jenkins job from the Jenkins console after the job's initial creation. These additional parameters are honored when builds are started for those Jenkins jobs.
+
+
+How you start a build for the Jenkins job determines how the parameters are set.
++
+* *oc start-build*. If you start the build using the `oc start-build` CLI command the environment variables specified in the `BuildConfig` take precedence. Any changes you make to the default  parameter values from the Jenkins console are ignored. 
++
+* *oc start-build -e*. If you start the build using the `oc start-build -e` command option, the values for the environment variables specified in the `BuildConfig` and with `-e` option take precedence.  If you specify an environment variable not listed in the `BuildConfig`, that variable will be added as a Jenkins job parameter definition. Any changes you make from the Jenkins console to the parameters corresponding to the environment variables are ignored.
```
Comment (gabemontero) [inline dev_guide/builds/build_strategies.adoc]: I think it is important to explicitly capture this aspect of the order of precedence:  env vars specified via `-e` can override env vars of the same name listed in the build config ... don't think we capture that currently.

File: dev_guide/builds/build_strategies.adoc [hunk 2]
```diff
@@ -568,24 +567,25 @@ jenkinsPipelineStrategy:
...
+Note the following details on the mappings between the Pipeline strategy `BuildConfig` environment variables and Jenkins job parameters:
+
+* When a Jenkins job is created or updated based on changes to the `BuildConfig`, any environment variables in the `BuildConfig` are mapped to Jenkins job parameters definitions. The default values for the Jenkins job parameters definitions are the current values of the associated environment variables.
+* You can add additional parameters (whose names differ from the names of the environment variables in the `BuildConfig`) 
+to the Jenkins job from the Jenkins console after the job's initial creation. These additional parameters are honored when builds are started for those Jenkins jobs.
+
+
+How you start a build for the Jenkins job determines how the parameters are set.
++
+* *oc start-build*. If you start the build using the `oc start-build` CLI command the environment variables specified in the `BuildConfig` take precedence. Any changes you make to the default  parameter values from the Jenkins console are ignored.
```
Comment (gabemontero) [inline dev_guide/builds/build_strategies.adoc]: How about this:

"If you start the build uisng `oc start-build` CLI command the current values for the environment variables specified in the `BuildConfig` take precedence over whatever default value the corresponding parameter in the Jenkins job may have."

It dawned on my that citing *how* they changed the value was irrelevant and *might* muddy the waters a little.

**Review by gabemontero — CHANGES_REQUESTED** [review]
It still was pretty close ... I've offered up a specific replacement in one spot, and they a specific addition where I deferred to you on how to exactly wordsmith it.

thanks

File: dev_guide/builds/build_strategies.adoc (hunk 1 above)
Comment (mburke5678) [inline dev_guide/builds/build_strategies.adoc]: @gabemontero How about:

If you start the build using the oc start-build -e command option, the values for the environment variables specified with the -e option take precedence. The variables not specified with in the -e option are taken from the BuildConfig.
Note:
 *  The environment varaibles specified with -e override environment variables of the same name listed in the BuildConfig.

 * If you specify an environment variable not listed in the BuildConfig, that variable will be added as a Jenkins job parameter definition.

Any changes you make from the Jenkins console to the parameters corresponding to the environment variables are ignored.

File: dev_guide/builds/build_strategies.adoc [hunk 3]
```diff
@@ -568,24 +567,37 @@ jenkinsPipelineStrategy:
...
+* *oc start-build*. If you start the build using the `oc start-build` CLI command, the current values for the environment variables specified in the BuildConfig take precedence over whatever default value the corresponding parameter in the Jenkins job has. 
++
+Any changes you make to the default  parameter values from the Jenkins console are ignored. 
+
+* *oc start-build -e*. If you start the build using the `oc start-build -e` command option, the values for the environment variables specified with the `-e` option take precedence. The variables not specified with in the `-e` option are taken from the `BuildConfig`. 
++
+--
+Note:
+
+* The environment varaibles specified with `-e` override environment variables of the same name listed in the `BuildConfig`.
```
Comment (gabemontero) [inline dev_guide/builds/build_strategies.adoc]: Should this sentence be moved from here and combined with the first sentence in line 589.  specifically:

"...specified with the `-e` option take precedence, overriding environment variables of the same name listed in the `BuildConfig`."

**Review by gabemontero — CHANGES_REQUESTED** [review]
Just one minor suggestion / question.  Pending the resolution of that one way or the other, I'm good to go.

File: dev_guide/builds/build_strategies.adoc (hunk 3 above)
Comment (mburke5678) [inline dev_guide/builds/build_strategies.adoc]: Yeah. I can move that sentence. I wanted to highlight, so I put it in a Note. But, the two notes aren't as related.

"If you start the build using the `oc start-build -e` command option, the values for the environment variables specified with the `-e` option take precedence and override any environment variables of the same name listed in the `BuildConfig`. The variables not specified with in the `-e` option are taken from the `BuildConfig`.

File: dev_guide/builds/build_strategies.adoc [hunk 4]
```diff
@@ -568,24 +567,35 @@ jenkinsPipelineStrategy:
...
+to the Jenkins job from the Jenkins console after the job's initial creation. These additional parameters are honored when builds are started for those Jenkins jobs.
+
+
+How you start a build for the Jenkins job determines how the parameters are set.
+
+* *oc start-build*. If you start the build using the `oc start-build` CLI command, the current values for the environment variables specified in the BuildConfig take precedence over whatever default value the corresponding parameter in the Jenkins job has. 
++
+Any changes you make to the default  parameter values from the Jenkins console are ignored. 
+
+* *oc start-build -e*. If you start the build using the `oc start-build -e` command option, the values for the environment variables specified with the `-e` option take precedence and override any environment variables of the same name listed in the `BuildConfig`. The variables not specified with in the `-e` option are taken from the `BuildConfig`.
```
Comment (gabemontero) [inline dev_guide/builds/build_strategies.adoc]: The last sentence seems a little off.   "... not specified with in the `--e` options ..."

How about:

"Any variables specified in the `BuildConfig` that are not referenced with the `--e` options are still taken."

**Review by gabemontero — CHANGES_REQUESTED** [review]
Apologies if I'm nit picking (or if I missed this previously) but one sentence I believe is still a bit off.

Let me know what you think.

Otherwise I'm good to go.

File: dev_guide/builds/build_strategies.adoc (hunk 4 above)
Comment (mburke5678) [inline dev_guide/builds/build_strategies.adoc]: Sounds fine to me. 
http://file.rdu.redhat.com/~mburke/3855/build_strategies.html#jenkins-pipeline-strategy-environment

**Comment by mburke5678** [comment]
@bfallonf can you take a look at my changes for RH style and any other editorial issues?

**Comment by mburke5678** [comment]
This is a follow-up PR related to https://github.com/openshift/openshift-docs/pull/3855/files

**Review by gabemontero — APPROVED** [review]
http://file.rdu.redhat.com/%7Emburke/3855/build_strategies.html#jenkins-pipeline-strategy-environment looks good

Thanks !

File: dev_guide/builds/build_strategies.adoc [hunk 5]
```diff
@@ -568,24 +567,35 @@ jenkinsPipelineStrategy:
...
+Note the following details on the mappings between the Pipeline strategy `BuildConfig` environment variables and Jenkins job parameters:
+
+* When a Jenkins job is created or updated based on changes to the `BuildConfig`, any environment variables in the `BuildConfig` are mapped to Jenkins job parameters definitions. The default values for the Jenkins job parameters definitions are the current values of the associated environment variables.
+* You can add additional parameters (whose names differ from the names of the environment variables in the `BuildConfig`) 
+to the Jenkins job from the Jenkins console after the job's initial creation. These additional parameters are honored when builds are started for those Jenkins jobs.
+
+
+How you start a build for the Jenkins job determines how the parameters are set.
+
+* *oc start-build*. If you start the build using the `oc start-build` CLI command, the current values for the environment variables specified in the BuildConfig take precedence over whatever default value the corresponding parameter in the Jenkins job has.
```
Comment (bfallonf) [inline dev_guide/builds/build_strategies.adoc]: For the above, can I suggest:

``oc start-build`: the current values for the environment variables specified in the BuildConfig take precedence over whatever default value the corresponding parameter in the Jenkins job has.`

Comment (bfallonf) [inline dev_guide/builds/build_strategies.adoc]: Note the markup around oc start-build

File: dev_guide/builds/build_strategies.adoc [hunk 6]
```diff
@@ -568,24 +567,35 @@ jenkinsPipelineStrategy:
...
+to the Jenkins job from the Jenkins console after the job's initial creation. These additional parameters are honored when builds are started for those Jenkins jobs.
+
+
+How you start a build for the Jenkins job determines how the parameters are set.
+
+* *oc start-build*. If you start the build using the `oc start-build` CLI command, the current values for the environment variables specified in the BuildConfig take precedence over whatever default value the corresponding parameter in the Jenkins job has. 
++
+Any changes you make to the default  parameter values from the Jenkins console are ignored. 
+
+* *oc start-build -e*. If you start the build using the xref:../../cli_reference/basic_cli_operations.html#start-build[oc start-build -e] command option, the values for the environment variables specified with the `-e` option take precedence and override any environment variables of the same name listed in the `BuildConfig`. Any variables specified in the BuildConfig that are not referenced with the `-e` options are still taken.
```
Comment (bfallonf) [inline dev_guide/builds/build_strategies.adoc]: Then similar to the comment above:

`oc start-build -e`: the values for the environment variables specified with the `-e` option take precedence and override any environment variables of the same name listed in the `BuildConfig`. Any variables specified in the BuildConfig that are not referenced with the `-e` options are still taken.

File: dev_guide/builds/build_strategies.adoc (hunk 5 above)
Comment (bfallonf) [inline dev_guide/builds/build_strategies.adoc]: Also, unless it's in backticks, BuildConfig should be spelled out "build configuration". But if you want to add the backticks to the above, that'll work as well.

File: dev_guide/builds/build_strategies.adoc [hunk 7]
```diff
@@ -568,24 +567,35 @@ jenkinsPipelineStrategy:
...
++
+--
+Note:
+
+* If you specify an environment variable not listed in the `BuildConfig`, that variable will be added as a Jenkins job parameter definition. 
+--
++
+Any changes you make from the Jenkins console to the parameters corresponding to the environment variables are ignored.  
+
+* *Jenkins console*. If you start the Jenkins job usng the Jenkins console, you set the environment variable though the Jenkins console as part of starting the build job.
```
Comment (bfallonf) [inline dev_guide/builds/build_strategies.adoc]: Then similar:

*Jenkins console*: set the environment variable though the Jenkins console as part of starting the build job.

File: dev_guide/builds/build_strategies.adoc [hunk 8]
```diff
@@ -568,24 +567,35 @@ jenkinsPipelineStrategy:
...
++
+Any changes you make to the default  parameter values from the Jenkins console are ignored. 
+
+* *oc start-build -e*. If you start the build using the xref:../../cli_reference/basic_cli_operations.html#start-build[oc start-build -e] command option, the values for the environment variables specified with the `-e` option take precedence and override any environment variables of the same name listed in the `BuildConfig`. Any variables specified in the BuildConfig that are not referenced with the `-e` options are still taken. 
++
+--
+Note:
+
+* If you specify an environment variable not listed in the `BuildConfig`, that variable will be added as a Jenkins job parameter definition. 
+--
```
Comment (bfallonf) [inline dev_guide/builds/build_strategies.adoc]: For the above, an actual Note box would be best:

[NOTE]
====
If you specify an environment variable not listed in the `BuildConfig`, that variable will be added as a Jenkins job parameter definition. 
====

**Comment by bfallonf** [comment]
@mburke5678 I made some comments. A few to fox some repetitiveness, and a couple on style-ey things.

Overall, good job :star:

File: dev_guide/builds/build_strategies.adoc (hunk 5 above)
Comment (mburke5678) [inline dev_guide/builds/build_strategies.adoc]: >>Note the markup around oc start-build (hard to see in Github)
What should the markup around oc start-build be?

Comment (bfallonf) [inline dev_guide/builds/build_strategies.adoc]: As it's a command, backticks.

(If I'm ever confused, I click on the "edit" button in a comment, and you can see the markup without Github rendering it :) )

**Comment by gabemontero** [comment]
I-Still-GTM  after the latest edits :-)

**Comment by vikram-redhat** [comment]
@mburke5678 - needs labels and milestones adding.

**Comment by mburke5678** [comment]
This is a follow up to https://github.com/openshift/openshift-docs/pull/3855.

**Comment by ahardin-rh** [comment]
@mburke5678 Are you still working on this, or can it be merged? Looks like it just needs to be squashed into one commit, then we can remove the to_followup label from the original PR when this merges.

**Pull request merged**